### PR TITLE
feat(dashboard): search for `wfRuns` by variable `name` & `value`

### PR DESCRIPTION
**Pull request by HazimAr**
image
image
image
image

### CODE REVIEW

**Comment by coltmcnealy-lh** [comment]
Does this support json search?

**Comment by HazimAr** [comment]
> Does this support json search?

No it doesn't, it only supports `str` search right now. I just wanted to get a bones version of the search that is easy to build on. Going to try to finish all the tickets on dashboard then support other var types.

**Comment by HazimAr** [comment]
Just found a bug with this, wont be able to fix until after my deployment.

The select shows all variables, it should only show variables that are `indexed`

**Comment by HazimAr** [comment]
Bug fixed now

File: dashboard/src/app/(authenticated)/(diagram)/wfSpec/[...props]/components/SearchVariableDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+      return { str: value }
+    case 'int':
+      return { int: parseInt(value) }
+    case 'bool':
+      return { bool: value.toLowerCase() === 'true' }
+    case 'double':
+      return { double: parseFloat(value) }
+    default:
+      return { str: value }
+  }
```
Comment (mijailr) [inline dashboard/src/app/(authenticated)/(diagram)/wfSpec/[...props]/components/SearchVariableDialog.tsx:146]: This should converted by the form instead, but this implementation is sufficient we should use a different approach if we plan to support `JSON_OBJ` and `JSON_ARR`

**Review by mijailr — APPROVED** [review]

**Pull request merged**